### ISSUE TITLE
Unexposed modules

### ISSUE THREAD

**Issue by rlpowell** [jgm/pandoc#3260]
So I have a project that needs a custom reader, so I made:

      lib/Text/Pandoc/Readers/TikiWiki.hs

and copied a bunch of stuff out of https://hackage.haskell.org/package/pandoc-1.18/docs/src/Text-Pandoc-Readers-TWiki.html

But it turns out that I needed some stuff that is set private, so I ended up cloning a couple of files and now I have:

```
lib/Text/Pandoc/Asciify.hs
lib/Text/Pandoc/Readers/TikiWiki.hs
lib/Text/Pandoc/Parsing.hs
```

That seems like a somewhat silly thing to have to do just to construct a custom reader; what's the idiomatic way to handle this?

BTW, if y'all want this reader, it's at https://github.com/rlpowell/hblog/blob/master/lib/Text/Pandoc/Readers/TikiWiki.hs ; feel free to do whatever.  I normally MIT license, but huge chunks are stolen from TWiki.hs, which I guess makes it GPL2?

**Comment by jgm** [comment]
This tracker is for bug reports and specific enhancement requests.  Questions like this should go to the pandoc-discuss mailing list.  Thanks!

But since we're here:  basically you're asking why a bunch of pandoc's modules are under Other-Modules rather than Exposed-Modules in pandoc.cabal, and in particular

                   Text.Pandoc.Writers.Shared,
                   Text.Pandoc.Asciify,
                   Text.Pandoc.MIME,
                   Text.Pandoc.Emoji,
                   Text.Pandoc.Parsing,
                   Text.Pandoc.UUID,
                   Text.Pandoc.ImageSize,
                   Text.Pandoc.Slides,
                   Text.Pandoc.Highlighting,

Basically, I've tried to expose what is needed for normal users of the library, and hide everything else (which makes API changes less frequent and allows us to change the internals more aggressively).

I wasn't really thinking of the needs of people writing custom readers, though, and you're right that a good case could be made for exporting these modules.

**Comment by jgm** [comment]
We should probably expose these under Text.Pandoc.Internal, where we can change them without bumping the version.

**Comment by rlpowell** [comment]
I just want to check: this is closed in master, but there's no release with this change yet, right?

**Comment by mb21** [comment]
No release, but you can try https://github.com/pandoc-extras/pandoc-nightly
